From ddd9da98fd86be3730747f1da921b22f2452062c Mon Sep 17 00:00:00 2001
From: Till Kamppeter <till.kamppeter@gmail.com>
Date: Sat, 5 Feb 2022 16:22:10 -0300
Subject: [PATCH] pdftopdf: Add 2% tolerance for input size larger than output
 page

When "print-scaling=auto" or "print-scaling=auto-fit" is used, the
input pages are scaled when they do not fit into the output page
size. Often input ad out page sizes are supposed to be equal, for
example both A4, but rounding errors could make the input considered
larger and unnecessarily scaled.

Therefore we add 2% of tolerance before considering an input page too
large.

(manually backported commit e541dc698f38aaa522f99e23b57c695da2981c29)
---
 NEWS                                  | 6 ++++++
 filter/pdftopdf/pdftopdf_processor.cc | 4 ++--
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/filter/pdftopdf/pdftopdf_processor.cc b/filter/pdftopdf/pdftopdf_processor.cc
index e297a246e..28643163f 100644
--- a/filter/pdftopdf/pdftopdf_processor.cc
+++ b/filter/pdftopdf/pdftopdf_processor.cc
@@ -195,8 +195,8 @@ bool processPDFTOPDF(PDFTOPDF_Processor &proc,ProcessingParameters &param) // {{
     for (int i = 0; i < (int)pages.size(); i ++)
     {
       PageRect r = pages[i]->getRect();
-      int w = r.width;
-      int h = r.height;
+      int w = r.width * 100 / 102; // 2% of tolerance
+      int h = r.height * 100 / 102;
       if ((w > param.page.width || h > param.page.height) &&
 	  (h > param.page.width || w > param.page.height))
       {
-- 
2.47.1

